browser-test/src/header.test.ts

```diff
@@ -1,56 +1,59 @@
+import {test, expect} from './support/civiform_fixtures'
 import {
-  createTestContext,
   loginAsTestUser,
   validateScreenshot,
   validateAccessibility,
+  logout,
 } from './support'
 
-test.describe('Header', () => {
-  const ctx = createTestContext(/* clearDb= */ false)
-
-  test('Not logged in, guest mode enabled', async () => {
-    const {page} = ctx
-    await validateScreenshot(
-      page.getByRole('navigation'),
-      'not-logged-in-guest-mode-enabled',
-    )
-  })
-
-  // TODO(#4360): add a "Not logged in, guest mode disabled" test once we
-  // can get to the programs page without logging in, for an entity without
-  // guest mode.
-
-  test('Logged in', async () => {
-    const {page} = ctx
-    await loginAsTestUser(page)
-    await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+test.describe('Header', {tag: ['@uses-fixtures']}, () => {
+  /**
+   * @todo (#4360) add a "Not logged in, guest mode disabled" test once we can get to the programs page without logging in, for an entity without guest mode.
+   */
+  test('Check screenshots and validate accessibility', async ({page}) => {
+    await test.step('Take a screenshot as a guest', async () => {
+      await logout(page)
+      await validateScreenshot(
+        page.getByRole('navigation'),
+        'not-logged-in-guest-mode-enabled',
+      )
+    })
+
+    await test.step('Take a screenshot as the test user', async () => {
+      await loginAsTestUser(page)
+      await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+    })
+
+    await test.step('Passes accessibility test', async () => {
+      await validateAccessibility(page)
+    })
   })
 
-  test('Passes accessibility test', async () => {
-    const {page} = ctx
-    await validateAccessibility(page)
-  })
-
-  test('Displays the government banner', async () => {
-    const {page} = ctx
-    expect(await page.textContent('section')).toContain(
-      'This is an official government website.',
+  test('Government banner', async ({page}) => {
+    const usaBannerLocator = page.getByRole('banner').locator('.usa-banner')
+    const usaBannerContentLocator = usaBannerLocator.locator(
+      '.usa-banner__content',
     )
-  })
-
-  test('Government banner expands when clicked and closes when clicked again', async () => {
-    const {page} = ctx
-    // The banner is initially closed
-    await expect(page.locator('.usa-banner__content')).toBeHidden()
-    // Click to expand the banner
-    await page.click('.usa-banner__button')
-
-    await expect(page.locator('.usa-banner__content')).toBeVisible()
-    await validateScreenshot(page.getByRole('navigation'), 'banner-expanded')
-    // Click again to close the banner
-    await page.click('.usa-banner__button')
-
-    await expect(page.locator('.usa-banner__content')).toBeHidden()
+    const usaBannerButtonLocator = usaBannerLocator.getByRole('button', {
+      name: "Here's how you know",
+    })
```

Comment:
Saving a reference to the parent-est locator and doing searches on that means we don't have to traverse the full page DOM every time. Makes it easier to read and can be better for performance.

Reply:
Using `page.getByRole` or other `page.getByXxx` methods is the preferred way to look up things, `getByRole` especially. This is because it's less fragile to look for things the way accessibility tools would.

Reply:
That's really good to know.  Can you help me understand why "name" is used?  I don't see a `name` attribute on this element.

Reply:
@melanie-exygy Name is the accessible name defined by w3c specs. Playwright has a good visual example [here](https://playwright.dev/docs/locators#locate-by-role) with more details.